examples/network/messaging.rs

```diff
@@ -97,46 +100,50 @@ impl<M: Send> Messaging<M> {
         let stop_rx = self.stop_rx.to_owned();
         let mut stop = false;
 
-        // TODO: `select_loop!` seems to really confuse Clippy.
+        // TODO: `select!` seems to really confuse Clippy.
         #[cfg_attr(
             feature = "cargo-clippy",
             allow(never_loop, if_let_redundant_pattern_matching, deref_addrof)
         )]
-        scope.spawn(move || {
+        scope.spawn(move |_| {
             let mut result = Ok(());
             // This loop forwards messages according to their metadata.
             while !stop && result.is_ok() {
-                select_loop! {
-                    recv(rx_from_algo, tm) => {
-                        match tm.target {
-                            Target::All => {
-                                // Send the message to all remote nodes, stopping at
-                                // the first error.
-                                result = txs_to_comms.iter()
-                                    .fold(Ok(()), |result, tx| {
-                                        if result.is_ok() {
-                                            tx.send(tm.message.clone())
-                                        } else {
-                                            result
-                                        }
-                                    }).map_err(Error::from);
-                            },
-                            Target::Node(i) => {
-                                result = if i < txs_to_comms.len() {
-                                    txs_to_comms[i].send(tm.message)
-                                        .map_err(Error::from)
-                                } else {
-                                    Err(Error::NoSuchTarget)
-                                };
+                select! {
+                    recv(rx_from_algo) -> tm => {
```

Comment:
Does the macro allow you to pattern match here? I.e. could you say `recv(rx_from_algo) -> Ok(tm)`?

Reply:
Ingenious! But that doesn't work.